### PR TITLE
[freetype] update toolchain and add layout

### PR DESCRIPTION
**Pull request by planetmarshall**
Specify library name and version:  **freetype/1.0**

* Update to new toolchain
* Add layout method

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/freetype/all/conanfile.py [hunk 1]
```diff
@@ -75,93 +73,98 @@ def requirements(self):
...
-        tools.replace_in_file(freetype_config, r"export LC_ALL", textwrap.dedent("""\
+        replace_in_file(self,freetype_config, r"%PKG_CONFIG%", r"/bin/false")  # never use pkg-config
+        replace_in_file(self,freetype_config, r"%prefix%", r"$conan_prefix")
+        replace_in_file(self,freetype_config, r"%exec_prefix%", r"$conan_exec_prefix")
+        replace_in_file(self,freetype_config, r"%includedir%", r"$conan_includedir")
+        replace_in_file(self,freetype_config, r"%libdir%", r"$conan_libdir")
+        replace_in_file(self,freetype_config, r"%ft_version%", r"$conan_ftversion")
+        replace_in_file(self,freetype_config, r"%LIBSSTATIC_CONFIG%", r"$conan_staticlibs")
+        replace_in_file(self,freetype_config, r"-lfreetype", libs)
+        replace_in_file(self,freetype_config, r"export LC_ALL", textwrap.dedent("""\
```
Comment (jwillikers) [inline recipes/freetype/all/conanfile.py:167]: Just need to add a space between the first and second arguments here.

File: recipes/freetype/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -1,10 +1,7 @@
...
 
-include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup(TARGETS)
-
-find_package(Freetype REQUIRED)
+find_package(Freetype CONFIG REQUIRED)
 
 add_executable(${PROJECT_NAME} test_package.c)
-target_link_libraries(${PROJECT_NAME} Freetype::Freetype)
+target_link_libraries(${PROJECT_NAME} freetype)
```
Comment (jwillikers) [inline recipes/freetype/all/test_package/CMakeLists.txt:7]:
```suggestion
target_link_libraries(${PROJECT_NAME} PRIVATE freetype)
```

File: recipes/freetype/all/test_v1_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package C)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+find_package(Freetype REQUIRED)
+
+add_executable(${PROJECT_NAME} ../test_package/test_package.c)
+target_link_libraries(${PROJECT_NAME} Freetype::Freetype)
```
Comment (jwillikers) [inline recipes/freetype/all/test_v1_package/CMakeLists.txt:10]:
```suggestion
target_link_libraries(${PROJECT_NAME} PRIVATE Freetype::Freetype)
```

File: recipes/freetype/all/test_package/conanfile.py [hunk 4]
```diff
@@ -1,18 +1,27 @@
...
         cmake.build()
 
     def test(self):
-        if not tools.cross_building(self):
-            bin_path = os.path.join("bin", "test_package")
+        if can_run(self):
+            bin_path = os.path.join(self.cpp.build.bindirs[0], "test_package")
             font_path = os.path.join(self.source_folder, "OpenSans-Bold.ttf")
-            self.run("{} {}".format(bin_path, font_path), run_environment=True)
+            self.run("{} {}".format(bin_path, font_path), env="conanrun")
```
Comment (jwillikers) [inline recipes/freetype/all/test_package/conanfile.py:27]:
```suggestion
            self.run(f"{bin_path} {font_path}", env="conanrun")
```

**Review by jwillikers — CHANGES_REQUESTED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

File: recipes/freetype/all/conanfile.py [hunk 5]
```diff
@@ -75,93 +73,98 @@ def requirements(self):
         if self.options.get_safe("with_brotli"):
             self.requires("brotli/1.0.9")
 
+    def layout(self):
+        basic_layout(self, src_folder="src")
```
Comment (prince-chrismc) [inline recipes/freetype/all/conanfile.py:77]: This should be using hte `cmake_layout`

Comment (prince-chrismc) [reply recipes/freetype/all/conanfile.py:77]: https://github.com/conan-io/conan-center-index/pull/13557

**Pull request merged**